routes/Wishlist.js

```diff
@@ -0,0 +1,65 @@
+import express from 'express';
+const router = express.Router();
+import requiredLogin from "../middleware/requiredLogin.js";
+import Wishlist from "../models/Wishlist.js";
+
+router.post("/addwishlist", requiredLogin, (req, res) => {
+
+    const { id, poster, title, date, media_type } = req.body;
+
+    if (!id || !title || !media_type) {
```

Comment:
It would be better if you create a separate middleware to check parameters so that here you could only focus on the actual requests.